url4e79c36700_JacenLakiir_Tivoo_tgz-pJ8-MainJ8/src/tivoo/output/HTMLBuilder.java

```diff
@@ -0,0 +1,382 @@
+    {
+        StringBuilder detailPageDirectory = new StringBuilder();
+        if (summaryPageFileName.contains("/"))
+        {
+            int lastFolder = summaryPageFileName.lastIndexOf("/");
+            detailPageDirectory.append(summaryPageFileName.substring(0, lastFolder + 1));
+            detailPageDirectory.append(DETAIL_PAGE_FOLDER);
+        }
+        else
+        {
+            detailPageDirectory.append(DETAIL_PAGE_FOLDER);
+        }
+        return detailPageDirectory.toString();
+    }
+    
+    private void buildSummaryPage (List<Event> eventList) throws IOException
+    {
+        if (mySummaryPageFileName.contains("/"))
+        {
+            File outputDirectory = new File(myDetailPageDirectory);
+            outputDirectory.mkdirs();
+        }
+        File summaryPage = new File(mySummaryPageFileName);
+        summaryPage.createNewFile();
+        writeSummaryPageHTML(summaryPage, eventList);
+    }
+    
+    private void buildDetailPages (List<Event> eventList) throws IOException
+    {
+        File detailsDirectory = new File(myDetailPageDirectory);
+        detailsDirectory.mkdirs();
+        for (Event currentEvent : eventList)
+        {
+            String detailPageURL = createDetailsPageURL(currentEvent);
+            File detailPage = new File(myDetailPageDirectory + "/" + detailPageURL);
+            boolean isNewFile = detailPage.createNewFile();
+            if (isNewFile)
+                writeDetailsPageHTML(currentEvent, detailPage);
+        }
+    }
+    
+    private void writeSummaryPageHTML (File summaryPage, List<Event> eventList) throws IOException
+    {
+        FileOutputStream fos = new FileOutputStream(summaryPage);
+        OutputStreamWriter out = new OutputStreamWriter(fos, "UTF-8");
+
+        Document doc = initializeHTMLDocument(getTitle(), "");
+        
+        String uniqueCSS = getUniqueCSS();
+        if (uniqueCSS.length() > 0)
+        {
+            doc.head.appendChild(insertCSS(uniqueCSS));
+        }
+        
+        writeHeader(doc);    
+        writeSummaryPageContent(doc, eventList);
+        writeFooter(doc);
+        
+        out.write(doc.write());
+        out.close();
+    }
+    
+    private void writeSummaryPageContent(Document doc, List<Event> eventList)
+    {
+        Div content = new Div().setCSSClass("content");
+        String title = getTitle();
+        content.appendChild(new H3().appendText(title));
+        Node view;
+        if (eventList.size() != 0)
+            view = buildView(eventList);
+        else
+            view = displayEmptyEventListWarning();
+        content.appendChild(view);
+        doc.body.appendChild(content);
+    }
+    
+    private P displayEmptyEventListWarning ()
+    {
+        return new P().appendText("No events to display.");
+    }
+    
+    private void writeDetailsPageHTML (Event currentEvent, File detailPage) throws IOException
+    {
+        FileOutputStream fos = new FileOutputStream(detailPage);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *fos*. The resource is referenced at lines 274, 281. The resource is closed at line 282. There are other execution paths that do not contain closure statements, for example, when Writer.write throws an exception. Ensure *fos* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).